3-enrich/stream-enrich/core/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/Enrich.scala

```diff
@@ -63,9 +63,14 @@ trait Enrich {
     val trackerSource: Either[String, (Option[Tracker[Id]], Source)] = for {
       config <- parseConfig(args)
       (enrichConfig, resolverArg, enrichmentsArg, forceDownload) = config
+      sourceSink = enrichConfig.streams.sourceSink.asInstanceOf[SourceSinkAgnosticConfig]
+      credentials <- (
+        sourceSink.aws.fold[Credentials](NoCredentials)(identity),
+        sourceSink.gcp.fold[Credentials](NoCredentials)(identity)
+      ).asRight
```

Comment:
Would be nice to add a unit test that checks that `credentials` gets the expected value depending on the configuration